reference-architecture/aws-ansible/playbooks/roles/rhsm-repos/tasks/main.yaml

```diff
@@ -1,11 +1,12 @@
 ---
 - name: disable unneeded repos
   command: subscription-manager repos --disable='*'
-  when: register_result | changed
+  when: register_result | changed and ansible_os_family == "RedHat"
```

Comment:
I think ansible_os_family includes CentOS, I think you want to test for `ansible_distribution not in ['Fedora', 'CentOS']` as well